resolwe_bio/tools/make_amplicon_table.py

```diff
@@ -0,0 +1,172 @@
+#!/usr/bin/env python2
+"""Create amplicon variant table."""
+from __future__ import absolute_import, division, print_function
+
+import argparse
+import json
+
+import HTSeq
+
+
+def parse_arguments():
+    """Parse command line arguments."""
+    parser = argparse.ArgumentParser(description="Parse Diff Exp output files")
+    parser.add_argument('master_file', help="Amplicon master file.")
+    parser.add_argument('amplicon_coverage', help='Per-amplicon mean coverage.')
+    parser.add_argument('-v', '--variants', nargs='+', required=True, help='snpEff output file')
+    return parser.parse_args()
+
+
+def _tsv_to_list(table_file, has_header=False, delimiter='\t', pick_columns=None):
+    """Parse *.tsv file into list."""
+    table_data = []
+    header = None
+    with open(table_file, 'r') as tfile:
+        if has_header:
+            header = next(tfile).strip().split(delimiter)
+            common_columns = [x for x in pick_columns if x in header]
+            if pick_columns:
+                # Find indexes of selected columns
+                temp_header = {col: i for i, col in enumerate(header)}
+                pick_indexes = [temp_header[col] for col in common_columns]
+                header = common_columns
+        for line in tfile:
+            line_content = line.strip().split(delimiter)
+            if pick_columns:
+                # Select only specific columns and order them as in pick_columns:
+                temp_line_content = {i: col for i, col in enumerate(line_content)}
+                line_content = [temp_line_content[i] for i in pick_indexes]
+            table_data.append(line_content)
+    return table_data, header
+
+
+def snp_url(snpid):
+    """Create url SNP ID."""
+    if snpid.startswith('rs'):
+        url = 'http://www.ncbi.nlm.nih.gov/snp/?term={}'.format(snpid)
+        pass
+    elif snpid.startswith('COSM'):
+        url = 'http://grch37-cancer.sanger.ac.uk/cosmic/mutation/overview?id={}'.format(snpid.lstrip('COSM'))
+    elif snpid.startswith('COSN'):
+        url = 'http://cancer.sanger.ac.uk/cosmic/ncv/overview?id={}'.format(snpid.lstrip('COSN'))
+    else:
+        return snpid
+    return url
+
+
+def main():
+    """Invoke when run directly as a program."""
+    args = parse_arguments()
+
+    amplicons = {}
+    amp_cov = {}
+    variants = []
+
+    with open(args.master_file) as master_file:
+        for line in master_file:
+            amplicon = line.strip().split('\t')
+
+            amplicons[amplicon[3]] = HTSeq.GenomicInterval(
+                amplicon[0],
+                int(amplicon[1]),
+                int(amplicon[2])
+            )
+
+    with open(args.amplicon_coverage) as ac:
+        for line in ac:
+            cov = line.strip().split('\t')
+            amp_cov[cov[0]] = float(cov[1])
+
+    for annot_var in args.variants:
```

Comment:
This argument passes one or multiple _variant files_ (2 in this use case). There is no issue with the character limit.